src/librustc/ty/mod.rs

```diff
@@ -2693,13 +2677,29 @@ fn def_span<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Span {
     tcx.hir.span_if_local(def_id).unwrap()
 }
 
+
+
+/// If the given def ID describes an item belonging to a trait,
+/// return the ID of the trait that the trait item belongs to.
+/// Otherwise, return `None`.
+fn trait_of_item<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Option<DefId> {
```

Comment:
I'm not sure exactly how many there really are left, though, because I skipped a bunch that were used in `librustc_resolve`. We also need to make a second macro (or another system) for all the ones that have `CrateNum` arguments instead of `DefId` arguments.